database/src/service/read.rs

```diff
@@ -0,0 +1,178 @@
+//! Database read thread-pool definitions and logic.
+
+//---------------------------------------------------------------------------------------------------- Import
+use std::task::{Context, Poll};
+
+use tokio::sync::oneshot;
+
+use crate::{
+    error::RuntimeError,
+    service::{request::ReadRequest, response::ReadResponse},
+    ConcreteEnv,
+};
+
+//---------------------------------------------------------------------------------------------------- Types
+/// The read response from the database reader thread.
+///
+/// This is an `Err` when the database itself errors
+/// (e.g, get() object doesn't exist).
+type Response = Result<ReadResponse, RuntimeError>;
```

Comment:
I think this is just meant to represent the successful state, so this should just be `ReadResponse` and then the Future should return Result<Response, Error>